### DISCUSSION TITLE
eslint-plugin-lit-a11y: flat config support & improvements for `d.ts`

### DISCUSSION THREAD

**Discussion by what1s1ove** [open-wc/open-wc#2798] [Ideas]
Hey guys! First of all, thank you for such a cool plugin and all you do!

Since ESLint 9 has been released, are there any plans to add a [Flat config](https://eslint.org/docs/latest/use/configure/configuration-files) and, **equally importantly**, add the `d.ts` files of the plugin?

<img width="1028" alt="image" src="https://github.com/open-wc/open-wc/assets/42296182/37aa5099-33b3-4337-8940-18c1dd961b11">

**Comment by thepassle** [comment]
Yeah, can you create a PR? 🙂

**Comment by favna** [comment]
Here is a setup that works for me:

```js
import { fixupPluginRules } from '@eslint/compat';
import { rules as litA11yRules, configs as litA11yConfigs } from 'eslint-plugin-lit-a11y';

/**
 * @type {import('@typescript-eslint/utils').TSESLint.FlatConfig.ConfigArray}
 */
const config = [
	{
		files: ['src/**/*.ts'],
		plugins: {
			'lit-a11y': {
				rules: fixupPluginRules(litA11yRules)
			}
		},
		rules: litA11yConfigs.recommended.rules
	},
]

export default config;
```

VSCode proof that it works:

![image](https://github.com/user-attachments/assets/3cde2fc7-7afd-4f3b-a1cb-37670bae78c3)

**Comment by caseywatts** [comment]
Here is how `eslint-plugin-jsx-a11y` updated their repo to support the new eslint flat config:

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/993
